database/src/progress.rs

```diff
@@ -41,29 +42,31 @@ impl Progress {
 
     pub async fn get_for_media_user(
         conn: &mut crate::Transaction<'_>,
-        uid: String,
+        uid: UserID,
         mid: i64,
     ) -> Result<Self, DieselError> {
         Ok(sqlx::query_as!(
             Progress,
-            "SELECT progress.* FROM progress
+            r#"SELECT id, user_id as "user_id: UserID", delta, media_id, populated FROM progress
             WHERE user_id = ?
-            AND media_id = ?",
+            AND media_id = ?"#,
             uid,
             mid
         )
         .fetch_optional(&mut *conn)
         .await?
         .unwrap_or(Self {
+            id: Default::default(),
             media_id: mid,
             user_id: uid,
-            ..Default::default()
```

Comment:
Keeping this is imo cleaner than specifying each field.

Reply:
My thought behind this is that I don't want to derive Default for`UserID`, as currently only valid IDs can be built, and deriving Default would give us invalid IDs. There isn't any way to partially derive Debug, so this is the closest I could get

Reply:
Sounds good.